src/ansys/pytwin/evaluate/twin_model.py

```diff
@@ -94,6 +96,127 @@ def _check_model_filepath_is_valid(self, model_filepath):
             raise self._raise_error(msg)
         return True
 
+    def _check_tbrom_model_filepath_is_valid(self, model_filepath):
+        """
+        Check if the resource directory path provided for the TbRom model instantiation is valid. Raise a
+        ``TwinModelError`` message if not.
+        """
+        if model_filepath is None:
+            msg = f"TbRom model cannot be instantiated with {model_filepath} as the resource directory path."
+            msg += "\nProvide a valid resource directory path to instantiate the ``TbRom`` object."
+            raise self._raise_error(msg)
+        if not os.path.exists(model_filepath):
+            msg = f"The provided resource directory does not exist: {model_filepath}."
+            msg += "\nProvide the correct filepath to instantiate the ``TbRom`` object."
+            raise self._raise_error(msg)
+        return True
+
+    def _check_rom_name_is_valid(self, rom_name):
+        """
+        Check if the rom name provided is part of the Twin's list of TbRom models. Raise a ``TwinModelError``
+        message if not.
+        """
+        if rom_name not in self.tbrom_names:
+            msg = f"The rom name provided {rom_name} is not part of the list of TbRom models {self.tbrom_names}."
+            msg += "\nCall this method with a valid rom name."
+            raise self._raise_error(msg)
+        return True
+
+    def _check_tbrom_input_field_dic_is_valid(self, tbrom_name, inputfieldsdic):
+        """
+        Check if the dictionary describing the input field snapshots files is valid and consistent with the Twin's
+        TBROM.
+        1st : check if input field name provided is valid
+        2nd : check if tbrom/twin have common inputs
+        3th : check if the provided snapshot path is valid
+        4th : check if the provided snapshot size is consistent with input field basis
+        Raise a ``TwinModelError`` message if not.
+        """
+        if self._check_rom_name_is_valid(tbrom_name):
+            tbrom = self._tbroms[tbrom_name]
+        for fieldname, snapshot in inputfieldsdic.items():
+            if fieldname not in tbrom.nameinputfields:
+                msg = (
+                    f"The field name provided {fieldname} is not part of the list of input field names "
+                    f"{tbrom.nameinputfields}."
+                )
+                msg += "\nProvide a valid field name to use this method."
+                raise self._raise_error(msg)
+            if not tbrom.hasinfmcs(fieldname):
+                msg = f"The tbrom {tbrom_name} has no common inputs with the Twin {self._model_name}."
+                msg += (
+                    "\nMake sure the TBROM has its mode coefficients inputs properly connected to the Twin " "inputs."
```

Comment:
... connected to the twin's ....
We don't want to capitalize what isn't a proper noun.